test/tests/functional/pbs_svr_dyn_res.py

```diff
@@ -685,52 +681,44 @@ def test_svr_dyn_res_permissions(self):
 
-        dyn_scr = '"foo !' + fp + '"'
-        # give write permission to user only
-        self.du.chmod(path=fp, mode=0744)
-        self.scheduler.set_sched_config({'server_dyn_res': dyn_scr},
-                                        validate=False)
-        self.check_access_log(fp, exist=False)
-
-        time.sleep(1)
-        match_from = int(time.time())
-        # give write permission to others
-        self.du.chmod(path=fp, mode=0746)
-        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'True'})
-        self.scheduler.log_match(fp + ' file has a non-secure file access',
-                                 starttime=match_from, existence=True,
-                                 max_attempts=10)
-
         # Create dynamic resource script in tmp directory and check
         # file permissions
-        fp = self.du.create_temp_file(body=scr_body, asuser=ROOT_USER)
-        # Add to filenames for cleanup
-        self.filenames.append(fp)
+        # du.create_dyn_res_script by default creates the script in /tmp
 
+        # give write permission to group and others
+        fp = self.du.create_dyn_res_script(scr_body, perm=0766)
+        self.filenames.append(fp)
         dyn_scr = '"foo !' + fp + '"'
         self.scheduler.set_sched_config({'server_dyn_res': dyn_scr},
                                         validate=False)
-
-        # give write permission to group and others
-        self.du.chmod(path=fp, mode=0766)
         self.check_access_log(fp)
 
         # give write permission to group
-        self.du.chmod(path=fp, mode=0764)
+        fp = self.du.create_dyn_res_script(scr_body, perm=0764)
+        self.filenames.append(fp)
+        dyn_scr = '"foo !' + fp + '"'
+        self.scheduler.set_sched_config({'server_dyn_res': dyn_scr},
+                                        validate=False)
         self.check_access_log(fp)
 
         # give write permission to others
-        self.du.chmod(path=fp, mode=0746)
+        fp = self.du.create_dyn_res_script(scr_body, perm=0746)
+        self.filenames.append(fp)
+        dyn_scr = '"foo !' + fp + '"'
+        self.scheduler.set_sched_config({'server_dyn_res': dyn_scr},
+                                        validate=False)
         self.check_access_log(fp)
 
         # give write permission to user only
-        self.du.chmod(path=fp, mode=0744)
+        fp = self.du.create_dyn_res_script(scr_body, perm=0744)
+        self.filenames.append(fp)
+        dyn_scr = '"foo !' + fp + '"'
+        self.scheduler.set_sched_config({'server_dyn_res': dyn_scr},
+                                        validate=False)
         self.check_access_log(fp, exist=False)
 
     def tearDown(self):
         # removing all files creating in test
-        for i in self.filenames:
-            if os.path.isfile(i):
-                os.remove(i)
-            elif os.path.isdir(i):
-                os.removedirs(i)
+        self.du.rm(path=self.filenames, sudo=True, force=True,
```

Comment:
you should probably also call PTL's tearDown() if you are going to replace it

Reply:
oops! I'll add it